tests/test_md.py

```diff
@@ -20,866 +19,891 @@
+    frozen_idxs = np.delete(np.arange(0, len(coords)), moving_idxs)
+    assert len(frozen_idxs) > 0
+    assert np.all(coords[frozen_idxs] == xs[-1][frozen_idxs])
 
-        intg = LangevinIntegrator(temperature, dt, friction, masses, seed)
+    # Verify that the bound potentials haven't been changed, as local md modifies potentials
+    for ref_val, bp in zip(reference_values, bps):
+        ref_du_dx, ref_u = ref_val
+        test_du_dx, test_u = bp.execute(coords, box)
+        np.testing.assert_array_equal(ref_du_dx, test_du_dx)
+        np.testing.assert_equal(ref_u, test_u)
 
-        intg_impl = intg.impl()
+    # Verify that running with a barostat doesn't change the results
+    group_idxs = get_group_indices(get_bond_list(unbound_potentials[0]))
+
+    pressure = 1.0
+
+    barostat = MonteCarloBarostat(coords.shape[0], pressure, temperature, group_idxs, 1, seed)
+    barostat_impl = barostat.impl(bps)
+
+    intg_impl = intg.impl()
+
+    ctxt = custom_ops.Context(coords, v0, box, intg_impl, bps, barostat=barostat_impl)
+
+    ctxt.setup_local_md(temperature, freeze_reference)
+    baro_xs, baro_boxes = ctxt.multiple_steps_local(num_steps, local_idxs, store_x_interval=x_interval, radius=radius)
+
+    assert baro_xs.shape == xs.shape
+    assert baro_boxes.shape == boxes.shape
+
+    # Results using a barostat should be identical.
+    np.testing.assert_array_equal(baro_xs, xs)
+    np.testing.assert_array_equal(baro_boxes, boxes)
+
+    # Verify that wrapping up the potentials in a summed potential is identical
+    summed_potential = SummedPotential(unbound_potentials, sys_params)
+    # Flatten the arrays so we can concatenate them.
+    summed_potential = summed_potential.bind(np.concatenate([p.reshape(-1) for p in sys_params]))
+    bp = summed_potential.to_gpu(precision=np.float32).bound_impl
+
+    intg_impl = intg.impl()
+
+    # Rerun with the summed potential
+    ctxt = custom_ops.Context(coords, v0, box, intg_impl, [bp])
+    ctxt.setup_local_md(constants.DEFAULT_TEMP, freeze_reference)
+    summed_pot_xs, summed_pot_boxes = ctxt.multiple_steps_local(
+        num_steps, local_idxs, store_x_interval=x_interval, radius=radius
+    )
+
+    assert summed_pot_xs.shape == xs.shape
+    assert summed_pot_boxes.shape == boxes.shape
 
-        ctxt = custom_ops.Context(coords, v0, box, intg_impl, bps)
+    # Results using a summed potential should be identical.
+    np.testing.assert_array_equal(summed_pot_xs, xs)
+    np.testing.assert_array_equal(summed_pot_boxes, boxes)
 
-        xs, boxes = ctxt.multiple_steps_local(num_steps, local_idxs, radius=radius)
 
-        assert xs.shape[0] == 1
+@pytest.mark.parametrize("freeze_reference", [True, False])
+def test_multiple_steps_local_entire_system(freeze_reference):
+    """Verify that running multiple_steps_local is valid even when consuming the entire system, IE radius ~= inf.
+
+    - Only a single particle, the reference, should not move
```

Comment:
"If `freeze_reference, ..."

Reply:
Fixed in https://github.com/proteneer/timemachine/pull/1014/commits/2ad3c38a2a5e185ed36eae984cff7a674d03252a